implementation/core/src/test/java/io/smallrye/faulttolerance/core/timeout/ScheduledExecutorTimeoutWatcherTest.java

```diff
@@ -74,21 +75,10 @@ public void notTimedOut() throws InterruptedException {
         assertThatWithin(100, "watch not running", () -> !watch.isRunning());
     }
 
-    // TODO waiting for a condition in a unit test shouldn't really be needed
-    //  ultimately, we should use Awaitility for waiting for a condition in a test, not home-grown utils like this
     private static void assertThatWithin(int timeoutMs, String message, Supplier<Boolean> test) {
-        long start = System.currentTimeMillis();
-        while (System.currentTimeMillis() - start < timeoutMs) {
-            try {
-                Thread.sleep(50);
-            } catch (InterruptedException e) {
-                throw new RuntimeException("Interrupted waiting for " + message);
-            }
-            if (test.get()) {
-                return;
-            }
-        }
-        fail(message + " not satisfied in " + timeoutMs + "ms");
+        await(message).pollInterval(Duration.ofMillis(50))
+                .atMost(Duration.ofMillis(timeoutMs))
+                .until(test::get);
```

Comment:
I think I'd just replace the calls to `assertThatWithin` with direct Awaitility calls, using `untilAsserted`. Something like

```java
await().pollInterval(Duration.ofMillis(10)).atMost(Duration.ofMillis(100)).untilAsserted(() -> {
    assertThat(watch.isRunning).as("watch not running").isFalse();
});
```